src/das-employer-favourites.ApplicationServices/Infrastructure/FatApiRepository.cs

```diff
@@ -1,36 +1,85 @@
+using Microsoft.Extensions.Logging;
+using Polly;
+using Polly.Retry;
 using SFA.DAS.Apprenticeships.Api.Client;
+using SFA.DAS.Apprenticeships.Api.Types.Exceptions;
+using SFA.DAS.Providers.Api.Client;
 
 namespace DfE.EmployerFavourites.ApplicationServices.Infrastructure
 {
     public class FatApiRepository : IFatRepository
     {
         private readonly IStandardApiClient _standardApiClient;
         private readonly IFrameworkApiClient _frameworkApiClient;
+        private readonly IProviderApiClient _providerApiClient;
+        private readonly ILogger<FatApiRepository> _logger;
+        private readonly AsyncRetryPolicy _retryPolicy;
 
-        public FatApiRepository(IStandardApiClient standardApiClient, IFrameworkApiClient frameworkApiClient)
+        public FatApiRepository(
+            IStandardApiClient standardApiClient,
+            IFrameworkApiClient frameworkApiClient,
+            IProviderApiClient providerApiClient,
+            ILogger<FatApiRepository> logger)
         {
             _standardApiClient = standardApiClient;
             _frameworkApiClient = frameworkApiClient;
+            _providerApiClient = providerApiClient;
+            _logger = logger;
+            _retryPolicy = GetRetryPolicy();
         }
 
-        public string GetApprenticeshipName(string apprenticeshipId)
+        public async Task<string> GetApprenticeshipNameAsync(string apprenticeshipId)
         {
             if (IsStandard(apprenticeshipId))
             {
-                return _standardApiClient.Get(apprenticeshipId).Title;
+                var standard = await _retryPolicy.ExecuteAsync(context => _standardApiClient.GetAsync(apprenticeshipId), new Context(nameof(GetProviderNameAsync)));
+
+                return standard.Title;
             }
-            else
+
+            var framework = await _retryPolicy.ExecuteAsync(context => _frameworkApiClient.GetAsync(apprenticeshipId), new Context(nameof(GetProviderNameAsync)));
+
+            return framework.Title;
+        }
+
+        public async Task<string> GetProviderNameAsync(int ukprn)
+        {
+            try
             {
-                return _frameworkApiClient.Get(apprenticeshipId).Title;
+                var provider = await _retryPolicy.ExecuteAsync(context => _providerApiClient.GetAsync(ukprn), new Context(nameof(GetProviderNameAsync)));
+
+                return provider.ProviderName;
+            }
+            catch(EntityNotFoundException ex)
+            {
+                _logger.LogError(ex, $"Fat Api didn't find a provider for {ukprn}");
+
+                return $"Unknown Provider ({ukprn})";
             }
         }
 
         private bool IsStandard(string apprenticeshipId)
         {
-            int standardId;
-            return int.TryParse(apprenticeshipId, out standardId);
+            return int.TryParse(apprenticeshipId, out int _);
+        }
+
+        private Polly.Retry.AsyncRetryPolicy GetRetryPolicy()
```

Comment:
Polly.Retry is not required here as using statement is present

Reply:
👍